packages/react-devtools-shared/src/utils.js

```diff
@@ -50,16 +49,9 @@ import {localStorageGetItem, localStorageSetItem} from './storage';
 import {meta} from './hydration';
 
 import type {ComponentFilter, ElementType} from './types';
-import type {LRUCache} from 'react-devtools-shared/src/types';
 
 const cachedDisplayNames: WeakMap<Function, string> = new WeakMap();
 
-// On large trees, encoding takes significant time.
-// Try to reuse the already encoded strings.
-const encodedStringCache: LRUCache<string, Array<number>> = new LRU({
```

Comment:
Why did we decide to remove this cache? It seems like the new encoding algorithm takes comparable time?

Reply:
I mentioned this in the PR description:
> Also based on the same benchmark, removing the encoding cache further speeds things up, so this PR also removes that.

If you look at the [original benchmark](https://jsbench.me/x6ku2p436b/1) you can see that the one with the cache is slower. However after thinking about that again this afternoon, I realized that this was b'c of how I setup the benchmark (the cache Map was being recreated each time). Setting it up in the _before_ block (so it's reused between test runs) makes the test with cache a little faster.

_That being said_, that's not really an accurate benchmark either because the cache will never have 100% hit rate after the first few encodings. I'm not really sure of the best way to measure this 😁

Reply:
If I make the original list _really long_ (several hundred strings, each one 50-70 characters), then the difference between the two implementations (cache and no cache) is _less_:

<table>
<tr>
	<td>Main branch
	<td>2848.22 ops/s ± 1.84%
	<td>30.68 % slower
<tr>
	<td>This PR
	<td>2343.15 ops/s ± 0.96%<
	<td>42.97 % slower
<tr>
	<td>This PR with caching
	<td>4108.83 ops/s ± 1.05%
	<td>Fastest
</table>

But even in this case, _with a lot of caching hits_, the difference in performance isn't that great. And it may be worth pointing out that this test uses native `Map` (which is probably faster than our `LRU` library, since that's also managing caching eviction).

So I'm tempted to just remove this caching layer for now.

Wondering if @gaearon has any thoughts. I _believe_ he added the cache.

Reply:
I'll re-add the cache for now, so the changes in this PR are minimal. We can always remove it later.